### PR TITLE
Added more type information

### PR DESCRIPTION
**Pull request by ssbarnea**
Adds type information to few more functions. Moves `FileType` to constants in order to be able to re-use in other places.

This change has no runtime effects, it only improves typing coverage.

Partial: #893

### CODE REVIEW

File: lib/ansiblelint/constants.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
 """Constants used by AnsibleLint."""
 import os.path
+import sys
+
+if sys.version_info >= (3, 8):
+    from typing import Literal  # pylint: disable-msg=E0611
```
Comment (webknjaz) [inline lib/ansiblelint/constants.py:6]: plz use readable message identifiers instead of some numbers

Comment (webknjaz) [reply lib/ansiblelint/constants.py:6]: also there's no `disable-msg` in docs, only `disable`

File: lib/ansiblelint/constants.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+else:
+    from typing_extensions import Literal
```
Comment (webknjaz) [inline lib/ansiblelint/constants.py:8]: The proper fallback idiom is
```suggestion

try:
    from typing import Literal  # pylint: disable-msg=E0611
except ImportError:
    from typing_extensions import Literal
```

Comment (ssbarnea) [reply lib/ansiblelint/constants.py:8]: Have you tried it locally? I suspect your approach may upset one of our linters. I am used to the same idiom, and if I remember well, there was an issue that forced me to do it the ugly way.

File: lib/ansiblelint/constants.py [hunk 3]
```diff
@@ -1,7 +1,17 @@
 """Constants used by AnsibleLint."""
 import os.path
+import sys
+
+# mypy needs this special condition or it may fail to run
+# https://github.com/python/typeshed/issues/3500#issuecomment-560958608
```
Comment (webknjaz) [inline lib/ansiblelint/constants.py:6]: This must mention py3.6 specifically because the link only shows the solution and not motivation.

Comment (ssbarnea) [reply lib/ansiblelint/constants.py:6]: Maybe you want me to also include instructions about pressing the left mouse button over the link and reading the entire page, maybe the reader will not know that to do?!

Comment (webknjaz) [reply lib/ansiblelint/constants.py:6]: link != motivation

**Pull request merged**